io/src/csv/read_cells.cpp

```diff
@@ -25,22 +25,22 @@ namespace {
 /// Comparator used for sorting cells. This sorting is one of the assumptions
 /// made in the clusterization algorithm
 struct cell_order {
-    bool operator()(const traccc::cell& lhs, const traccc::cell& rhs) const {
-        if (lhs.module_link != rhs.module_link) {
-            return lhs.module_link < rhs.module_link;
-        } else if (lhs.channel1 != rhs.channel1) {
+    bool operator()(const traccc::io::csv::cell& lhs,
+                    const traccc::io::csv::cell& rhs) const {
+        if (lhs.channel1 != rhs.channel1) {
```

Comment:
This is only equivalent if the cells are guaranteed to be on the same module; did you ensure this?

Reply:
Very good catch! :smile: I did this on purpose. Because the CSV reading code indeed, by construction, never compares cells coming from different modules using this function. So the `module_link` comparison earlier was not actually necessary.

At the same time, I was pondering about this myself as well. Since while the current code is absolutely correct like this, the very negligible performance loss incurred by checking the module ID could possibly be worth it for avoiding future problems, when the file reading logic may change. :thinking: